src/generator/AutoRest.Go/Templates/MethodTemplate.cshtml

```diff
@@ -229,3 +211,59 @@ func (client @(Model.Owner)) @(Model.ResponderMethodName)(resp *http.Response) (
     @:return
 @:}
 }
+
+@if (Model.IsPageable && !Model.IsNextMethod && Model.ListElement != null)
+{
+@:@EmptyLine
+@:// @(Model.ListCompleteMethodName) gets all elements from the list without paging.
+@:func (client @(Model.Owner)) @(Model.ListCompleteMethodName) @(Model.MethodParametersSignatureComplete) @(Model.MethodReturnSignatureComplete) {
+    @:resultChan := make(chan @((Model.ListElement.ModelType as SequenceTypeGo).GetElement))
+    @:errChan := make(chan error, 1)
+    @:go func() {
+        @:var err error
+        @:defer func() {
+            @:errChan <- err
```

Comment:
I think you want to change this to write to errChan only if err is not nil; this will allow the select on channels pattern to work properly (I made the same change in the generator for LROs).